app/process/handler/consensus-message/consensus-message.go

```diff
@@ -85,33 +112,106 @@ func (cmh ConsensusMessageHandler) handlePayload(payload []byte) error {
 		return errors.New(fmt.Sprintf("[%s] - Address is not valid - [%s]", m.TransactionId, address.String()))
 	}
 
-	messages, err := cmh.repository.GetTransaction(m.TransactionId, ethSig)
-	if err != nil {
+	mes, err := cmh.repository.GetTransaction(m.TransactionId, ethSig, hexHash)
+	if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
 		return errors.New(fmt.Sprintf("Failed to retrieve messages for TxId [%s], with signature [%s]. - [%s]", m.TransactionId, ethSig, err))
```

Comment:
`Failed to retrieve transaction message with TxID [%s], Signature [%s]. Error [%s].`